File: weaviate-admin-api/app/models/schema.py
from pydantic import BaseModel
from typing import List, Dict, Optional, Any


class Property(BaseModel):
    """Schema property."""
    name: str
    dataType: List[str]
    indexed: Optional[bool] = True


class ClassSchema(BaseModel):
    """Class schema information."""
    name: str
    description: Optional[str] = None
    properties: List[Dict[str, Any]]
    vectorConfig: Optional[Dict[str, Any]] = None
    objectCount: Optional[int] = 0


class SchemaResponse(BaseModel):
    """Schema response with all classes."""
    classes: List[ClassSchema]

File: weaviate-admin-ui/src/api/schema.ts
import apiClient from './client';
import { SchemaResponse, ClassSchema } from '../types';

/**
 * Get all classes in the schema
 * @param projectId Optional project_id to filter object counts by. If not provided, uses user's project_id from token.
 */
export const getSchema = async (projectId?: number | 'all' | null): Promise<SchemaResponse> => {
  const params: any = {};
  if (projectId && projectId !== 'all') {
    params.project_id = projectId;
  }
  const response = await apiClient.get<SchemaResponse>('/schema', { params });
  return response.data;
};

/**
 * Get specific class schema
 * @param className The name of the class
 * @param projectId Optional project_id to filter object count by. If not provided, uses user's project_id from token.
 */
export const getClassSchema = async (className: string, projectId?: number | 'all' | null): Promise<ClassSchema> => {
  const params: any = {};
  if (projectId && projectId !== 'all') {
    params.project_id = projectId;
  }
  const response = await apiClient.get<ClassSchema>(`/schema/${className}`, { params });
  return response.data;
};

